Source/WebCore/platform/graphics/mac/controls/ControlMac.mm

```diff
@@ -211,8 +249,15 @@ static void drawCellOrFocusRing(GraphicsContext& context, const FloatRect& rect,
             // For slider cells, draw only the knob.
             [(NSSliderCell *)cell drawKnob:rect];
         } else {
-            [cell drawWithFrame:rect inView:view];
-            [cell setControlView:nil];
+            // Check if we can draw the NSCell without the fake NSView.
+            if ([NSCell instancesRespondToSelector:@selector(_setFallbackBackingScaleFactor:)]) {
```

Comment:
could have been `[cell respondsToSelector:...]` instead. Also you can cache it.

Reply:
And maybe factor it out into a static? "supportsViewlessCells()" or something, so that it's not so mysterious why you're checking something with a wildly different name

Reply:
I added the static `supportsViewlessCells()` and cached the value so I have to use `[NSCell instancesRespondToSelector:...]` but just for a single time.